### DISCUSSION TITLE
Make `Source`s callable as simple functions instead of with `.next()`

### DISCUSSION THREAD

**Discussion by Evertt** [state-adapt/state-adapt#71] [Ideas]
Since `onclick={() => someEventSource.next()}` is very often used in components, why not allow for the removal of some boilerplate by letting `someEventSource()` be called as if it were just a simple function? Then one could simply write `onclick={someEventSource}`.

**Comment by mfp22** [comment]
I love this idea. My only hesitation is that it would be different from RxJS Subject. `Source` wraps the values in `Action`s so occasionally I've switched back to `Subject` if I wanted to do some processing before giving it to the store without having to access `action.payload`. This is a slightly rough edge with StateAdapt. Calling as a function would be another thing you'd have to change if you switched between them. Maybe someday `action.type` can be a property of the `Source` instead, except then you couldn't use any RxJS operator or it would get rid of that property. 

Also if you follow the RxJS convention of `$` instead of `Source`, these are about the same number of characters: 
```typescript
onClick={someEvent$.next}
onClick={someEventSource}
```

But I'm wondering if RxJS allows `.next` to be passed like that. I tried it and it doesn't work. 

```typescript
onClick={() => someEvent$.next()}
onClick={someEventSource}
```

Yeah. I hate boilerplate so I'm definitely thinking about this. 

One other benefit is that you can clearly see that there are no side-effects to be performed by calling `next`. One of the big points of StateAdapt is to try to avoid callback functions. When I see `onClick={someEventSource}` maybe at a glance I can't see that it's not a function full of imperative changes.
